pkg/kubelet/probe.go

```diff
@@ -35,50 +37,83 @@ import (
 
 const maxProbeRetries = 3
 
-// probeContainer probes the liveness/readiness of the given container.
+// Prober helps to check the liveness/readiness of a container.
+// TODO(yifan): Replace the concrete type with interface later.
+type Prober struct {
+	exec   execprobe.ExecProber
+	http   httprobe.HTTPProber
+	tcp    tcprobe.TCPProber
+	runner dockertools.ContainerCommandRunner
+
+	readinessManager *kubecontainer.ReadinessManager
+	refManager       *kubecontainer.RefManager
+	recorder         record.EventRecorder
+}
+
+// NewProber creates a Prober, it takes a command runner and
+// several container info managers.
+func NewProber(
```

Comment:
Another way to work around this is to define an interface in kubelet/container:
``` go
type prober interface {
    Probe()
}
```

And then we can pass the kubelet as this interface to the runtime without the circular import problem.

But I feel it's cleaner to just move it to kubelet/prober. What's your opinion?